src/curvefit/initializer/initializer.py

```diff
@@ -0,0 +1,352 @@
+import numpy as np
+from typing import List
+
+from curvefit.solvers.solvers import Solver
+from curvefit.core.effects2params import unzip_x
+
+
+class PriorInitializerComponent:
+    """
+    {begin_markdown PriorInitializerComponent}
+
+    # `curvefit.initializer.initializer.PriorInitializerComponent`
+    ## A component of a prior initializer for a model
+
+    For a given solver, model and parameter set there are a number of strategies to get
+    information to inform a prior for a later model fit. This base class is used to build
+    prior initializer components, that is, objects that extract information from models that have already been run
+    to get well-informed priors for later runs.
+
+    Prior initializer components can have two types: priors that are extracted from the analysis of a joint model run
+    (one model run with > 1 group), and priors that are extracted from the analysis of a bunch of
+    individual models runs (many models runs with 1 group each). See the documentation on
+    [`JointPriorInitializerComponent`](JointPriorInitializerComponent.md) and
+    [`IndividualPriorInitializerComponent`](IndividualPriorInitializerComponent.md).
+
+    ## Attributes
+
+    - `self.component_type (str, None)`: type of component, overridden in subclasses ("joint" or "individual")
+
+    ## Methods
+
+    ### `_extract_prior`
+    A method to extract prior information from a solver or a list of solvers. Overridden in subclasses.
+
+    - `solver (List[Solver], curvefit.core.solver.Solver)`: solver or list of
+        solvers that have fit models to data using a parameter set
+
+    ### `_update_parameter_set`
+    A method to update a parameter set given information that was extracted about the priors from the solvers.
+
+    - `solver (List[Solver], curvefit.core.solver.Solver)`: see above
+    - `parameter_set_prototype (curvefit.core.parameter.ParameterSet)`: a parameter set that will be updated
+        with new prior information
+
+    {end_markdown PriorInitializerComponent}
+    """
+    def __init__(self):
+        self.component_type = None
+
+    def _extract_prior(self, solver):
+        pass
+
+    def _update_parameter_set(self, parameter_set_prototype, solver):
+        pass
+
+
+class JointPriorInitializerComponent(PriorInitializerComponent):
+    """
+    {begin_markdown JointPriorInitializerComponent}
+
+    # `curvefit.initializer.initializer.JointPriorInitializerComponent`
+    ## A joint prior initializer component for an initializer for a model
+
+    See [`PriorInitializerComponent`](PriorInitializerComponent.md).
+
+    ## Attributes
+
+    - `self.component_type (str)`: "joint", which is the type of component
```

Comment:
self.initializer_type?

Reply:
@aksholokhov @zhengp0 actually this should be `self.component_type` -- I've changed in the code where inconsistent! Good catch. See commit bf548d7.